### DISCUSSION TITLE
can we delete old json files in the transaction log

### DISCUSSION THREAD

**Discussion by djouallah** [delta-io/delta-rs#1555] [General]
currently I insert 288 files per day, which generate 288 json files in the transaction log, is there a way to delete old json ?

**Comment by dennyglee** [comment]
Hey @djouallah by default, the log retention is 30 days.  You can certainly modify it for your scenario via the `delta.logRetentionDuration` property which you set with the `ALTER TABLE SET TBLPROPERTIES` SQL method ([reference](https://docs.delta.io/latest/delta-utility.html#remove-files-no-longer-referenced-by-a-delta-table))

**Reply by yefetBenTili** [reply]
Is there a way to edit this configuration within delta-rs and not Databricks?

**Reply by ion-elgreco** [reply]
@yefetBenTili currently it's only possible to set during a create table operation or write operation when table does not exist yet. We still need to build this alter tbl properties operation in Rust and then expose it to Python.

**Comment by djouallah** [comment]
Thanks does this works with Delta Rust ?

**Reply by rtyler** [reply]
Unfortunately right now I don't believe delta-rs does any transaction log culling on reads or writes. I think that would be a good improvement though!

**Comment by yefetBenTili** [comment]
I had a similar issue and thanks to @ion-elgreco who pointed me to `cleanup_metadata()` but it requires an explicit call somewhere in your logic

**Reply by djouallah** [reply]
@ion-elgreco is there a way to change configuration.logRetentionDuration to something shorter than 30 days

**Reply by ion-elgreco** [reply]
It's only possible to define a table configuration during create at the moment. In the next couple weeks I'll start working on alter table properties, I've been pushing this one back a lot due to shifting interests haha

**Reply by djouallah** [reply]
@ion-elgreco last question, I notice that with the new version, the checkpoint parquet files no longer uses rle_dictionnary encoding, is this by design, I hope this does not change

**Reply by ion-elgreco** [reply]
@djouallah do you have an example of before and after

**Reply by djouallah** [reply]
never mind, I think it is triggered only with a massive number of logs entry, basically when the checkpoint become big enough

**Reply by djouallah** [reply]
Sorry to bother again but can you give an example how to define retention period at the table creation not very clear from the documentation

**Reply by djouallah** [reply]
@ion-elgreco Sorry to bother again but can you give an example how to define retention period at the table creation not very clear from the documentation

**Reply by ion-elgreco** [reply]
@djouallah 
I believe you should do it like this:

`DeltaTable.create(configuration = {"delta.logRetentionDuration": "<value>"})`
